woocommerce/woocommerce
# Release JS Packages: Execute script from GH Action

### All Submissions:






### Changes proposed in this Pull Request:

Use a Github Action to execute release prep tasks (changelog, version bump) script and create a pull request from the resulting diff.

As part of https://github.com/woocommerce/woocommerce/issues/33319

### How to test the changes in this Pull Request:

1. In the Actions tab, find [the `prepare-package-release` workflow](https://github.com/woocommerce/woocommerce/actions/workflows/prepare-package-release.yml)
2. Find the "Run Workflow" button
<img width="322" alt="image" src="https://user-images.githubusercontent.com/1922453/175900732-d8c74b37-4370-4c10-a8db-1c38c5282d7c.png">

4. Use this branch, `add/prep-package-release-workflow` and leave the default "All" input `-a` to prep all eligible packages
<img width="414" alt="image" src="https://user-images.githubusercontent.com/1922453/175900949-27001243-7523-4274-86f1-bd260f097edc.png">

5. Run the workflow and watch it succeed. There will be a pull request created similar to [this one](https://github.com/woocommerce/woocommerce/pull/33578). You should be pinged as a reviewer on the pull request.
6. Make sure all relevant information one might want is included in the pull request.
7. Repeat the process by kicking off the same workflow. This time specify a single package like `@woocommerce/components`. The same PR will update to reflect the change.
8. Close your PR

> **Note** Other changes

* Better error surfacing in `try/catch` blocks
* Fix a bad pattern matching error where paths including multiple `woocommerce` folders weren't recognized.

> **Note** Incorrect epilogue

If you look at the resulting PR's diff, the footer at the bottom of the changelog is changed to reference `woocommerce-admin/blob/main/packages/components/CHANGELOG.md)`. This is something I'll tackle on the next PR.


### Other information:




### FOR PR REVIEWER ONLY:
